PerfettoCds/Pipeline/PerfettoSourceParser.cs

```diff
@@ -0,0 +1,216 @@
+﻿using Microsoft.Performance.SDK.Extensibility.SourceParsing;
+using Microsoft.Performance.SDK.Processing;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using Perfetto.Protos;
+using Microsoft.Performance.SDK;
+using PerfettoCds.Pipeline.Events;
+using System.Linq;
+
+namespace PerfettoCds
+{
+    public sealed class PerfettoSourceParser : ISourceParser<PerfettoSqlEvent, PerfettoSourceParser, string>
+    {
+        public string Id => PerfettoPluginConstants.ParserId;
+
+        public Type DataElementType => typeof(PerfettoSqlEvent);
+        public Type DataContextType => typeof(PerfettoSourceParser);
+        public Type DataKeyType => typeof(string);
+
+        public int MaxSourceParseCount => 1;
+
+        // Information about this data source the SDK requires for building tables
+        private DataSourceInfo dataSourceInfo { get; set; }
+
+        public DataSourceInfo DataSourceInfo => this.dataSourceInfo;
+
+        // For WPA's loading bar
```

Comment:
Don't mention WPA as it could be another UI. Say UI in replace of WPA anywhere in code